src/lib.rs

```diff
@@ -848,10 +860,117 @@ impl LineSpans for alloc::string::String {
     }
 }
 
+#[derive(Debug, Clone)]
+/// Owning pointer to a string that provides O(1) access to line slices
+///
+/// ```
+/// use line_span::CachedLines;
+/// let cache = CachedLines::without_ending(String::from("hello\nworld"));
+/// assert_eq!(&cache[0], "hello");
+/// assert_eq!(&cache[1], "world");
+/// ```
+///
+/// Slices might or might not include line breaks depending on a function used to construct it
+///
+/// - [`without_ending`](CachedLines::without_ending)
+/// - [`with_ending`](CachedLines::with_ending)
+///
+///
+#[cfg(feature = "alloc")]
+pub struct CachedLines {
+    content: alloc::string::String,
+    splits: alloc::vec::Vec<Range<usize>>,
```

Comment:
Maybe instead we could have a:

```rust
pub struct LineSpanWithoutText {
    start: usize,
    end: usize,
    ending: usize,
}
```

and then:

```suggestion
    splits: alloc::vec::Vec<LineSpanWithoutText>,
```

That way we could replace `with_ending()` and `without_ending()` with `new()`

Reply:
It's been a while but as far as I remember the reasoning was that you are very unlikely to use both representations at once and having just one representation gives `Index` and iterator goodies.